pkg/yang/camelcase.go

```diff
@@ -28,30 +28,38 @@ func isASCIIDigit(c byte) bool {
 	return '0' <= c && c <= '9'
 }
 
-// CamelCase returns the CamelCased name.
-// If there is an interior underscore or dash followed by a lower case letter,
-// drop the underscore or dash and convert the letter to upper case.  There is a
-// remote possibility of this rewrite causing a name collision, but it's so
-// remote we're prepared to pretend it's nonexistent - since the C++ generator
-// lowercases names, it's extremely unlikely to have two fields with different
-// capitalizations.  In short, _my_field_name_2 becomes XMyFieldName_2.
+// CamelCase returns a CamelCased name for a YANG identifier.
+// Currently this supports the output being used for a Go or proto identifier.
+// Dash and dot are first converted to underscore, and then any underscores
+// before a lower-case letter are removed, and the letter converted to upper-case.
+// The first letter is always upper-case in order to be an exported name in Go.
+// There is a remote possibility of this rewrite causing a name collision, but
+// it's so remote we're prepared to pretend it's nonexistent - since the C++
+// generator lowercases names, it's extremely unlikely to have two fields with
+// different capitalizations. In short, _my_field-name_2 becomes XMyFieldName_2.
+// If the input does not conform to the YANG identifier specification
+// (https://tools.ietf.org/html/rfc7950#section-6.2), then there is no
```

Comment:
The new "undefined" behaviour is actually the same as before (i.e. treat them as lower-case characters). I've added tests below and changed the comment from undefined behaviour to this description.

Reply:
Let me know if this is now resolved.

Reply:
Ok, I think I understand -- unrecognized characters are essentially passthrough; it might manifest itself as compile errors if we end up with Go identifiers that are invalid.